In [1]:
import pandas as pd

In [8]:
fund_master = pd.read_csv(r"C:\Users\WINDOWS\OneDrive\Documents\Mutual_Fund_Analytics_Project\data\raw\01_fund_master.csv")

In [11]:
fund_master.head()

,amfi_code,fund_house,scheme_name,category,sub_category,plan,launch_date,benchmark,expense_ratio_pct,exit_load_pct,min_sip_amount,min_lumpsum_amount,fund_manager,risk_category,sebi_category_code
0,119551,SBI Mutual Fund,SBI Bluechip Fund - Regular Plan - Growth,Equity,Large Cap,Regular,2006-02-14,NIFTY 100 TRI,1.54,1.0,500,1000,Sohini Andani,Moderate,EC01
1,119552,SBI Mutual Fund,SBI Bluechip Fund - Direct Plan - Growth,Equity,Large Cap,Direct,2013-01-01,NIFTY 100 TRI,0.66,1.0,500,1000,Sohini Andani,Moderate,EC01
2,119598,SBI Mutual Fund,SBI Small Cap Fund - Regular Plan - Growth,Equity,Small Cap,Regular,2009-09-09,BSE 250 SmallCap TRI,1.43,1.0,500,1000,R. Srinivasan,Very High,EC03
3,119599,SBI Mutual Fund,SBI Small Cap Fund - Direct Plan - Growth,Equity,Small Cap,Direct,2013-01-01,BSE 250 SmallCap TRI,0.72,1.0,500,1000,R. Srinivasan,Very High,EC03
4,119120,SBI Mutual Fund,SBI Magnum Gilt Fund - Regular Plan - Growth,Debt,Gilt,Regular,2000-12-30,CRISIL Dynamic Gilt Index,0.77,0.0,500,1000,Dinesh Ahuja,Low,DC02


In [13]:
fund_master.shape

(40, 15)

In [12]:
fund_master.columns

Index(['amfi_code', 'fund_house', 'scheme_name', 'category', 'sub_category',
       'plan', 'launch_date', 'benchmark', 'expense_ratio_pct',
       'exit_load_pct', 'min_sip_amount', 'min_lumpsum_amount', 'fund_manager',
       'risk_category', 'sebi_category_code'],
      dtype='object')

In [14]:
fund_master.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 15 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   amfi_code           40 non-null     int64  
 1   fund_house          40 non-null     object 
 2   scheme_name         40 non-null     object 
 3   category            40 non-null     object 
 4   sub_category        40 non-null     object 
 5   plan                40 non-null     object 
 6   launch_date         40 non-null     object 
 7   benchmark           40 non-null     object 
 8   expense_ratio_pct   40 non-null     float64
 9   exit_load_pct       40 non-null     float64
 10  min_sip_amount      40 non-null     int64  
 11  min_lumpsum_amount  40 non-null     int64  
 12  fund_manager        40 non-null     object 
 13  risk_category       40 non-null     object 
 14  sebi_category_code  40 non-null     object 
dtypes: float64(2), int64(3), object(10)
memory usage: 4.8+ KB


In [15]:
fund_master.describe()

,amfi_code,expense_ratio_pct,exit_load_pct,min_sip_amount,min_lumpsum_amount
count,40.000000,40.000000,40.000000,40.0,40.000000
mean,120247.000000,1.237000,0.812500,500.0,1277.500000
std,14534.998667,0.386584,0.387091,0.0,1082.847031
min,100016.000000,0.550000,0.000000,500.0,100.000000
25%,118632.750000,0.787500,1.000000,500.0,1000.000000
50%,119551.500000,1.425000,1.000000,500.0,1000.000000
75%,120842.250000,1.540000,1.000000,500.0,1000.000000
max,149324.000000,1.640000,1.000000,500.0,5000.000000


In [16]:
fund_master["fund_house"].value_counts()

fund_house
SBI Mutual Fund             5
HDFC Mutual Fund            5
ICICI Prudential MF         5
Nippon India MF             5
Kotak Mahindra MF           4
Axis Mutual Fund            4
Aditya Birla Sun Life MF    3
UTI Mutual Fund             3
Mirae Asset MF              3
DSP Mutual Fund             3
Name: count, dtype: int64

The dataset contains schemes from 10 major Asset Management Companies (AMCs), with SBI Mutual Fund, HDFC Mutual Fund, ICICI Prudential MF, and Nippon India MF contributing the highest number of schemes (5 each).

In [18]:
fund_master["category"].value_counts()

category
Equity    34
Debt       6
Name: count, dtype: int64

In [19]:
fund_master["sub_category"].value_counts()

sub_category
Large Cap          14
Mid Cap             7
Small Cap           6
Liquid              3
Gilt                2
Flexi Cap           2
Value               1
Short Duration      1
Index/ETF           1
Index               1
Large & Mid Cap     1
ELSS                1
Name: count, dtype: int64

Large Cap funds dominate the dataset.

This suggests the project focuses on mainstream retail investment products.

In [20]:
fund_master["risk_category"].value_counts()

risk_category
Moderate           16
High                8
Very High           6
Low                 6
Moderately High     4
Name: count, dtype: int64

In [21]:
nav_history = pd.read_csv(r"C:\Users\WINDOWS\OneDrive\Documents\Mutual_Fund_Analytics_Project\data\raw\02_nav_history.csv")

In [22]:
nav_history.head()

,amfi_code,date,nav
0,119551,2022-01-03,54.3856
1,119551,2022-01-04,54.3474
2,119551,2022-01-05,54.6869
3,119551,2022-01-06,55.4550
4,119551,2022-01-07,55.3692


In [23]:
print(nav_history.shape)

(46000, 3)


In [24]:
print(nav_history.dtypes)

amfi_code      int64
date          object
nav          float64
dtype: object


In [25]:
nav_history["amfi_code"].nunique()

40

In [27]:
nav_history["date"].min()

'2022-01-03'

In [28]:
nav_history["date"].max()

'2026-05-29'

In [29]:
nav_history.isnull().sum()

amfi_code    0
date         0
nav          0
dtype: int64

In [30]:
master_codes = set(fund_master["amfi_code"])

nav_codes = set(nav_history["amfi_code"])

missing_codes = master_codes - nav_codes

print("Missing Codes:", missing_codes)

print("Number of Missing Codes:", len(missing_codes))

Missing Codes: set()
Number of Missing Codes: 0


In [ ]:

1. The fund_master dataset contains 40 mutual fund schemes with 15 attributes.

2. The nav_history dataset contains 46,000 daily NAV records covering the period from 03-Jan-2022 to 29-May-2026.

3. No missing values were found in fund_master or nav_history.

4. All 40 AMFI scheme codes present in fund_master were successfully found in nav_history.

5. AMFI code validation returned zero missing codes, confirming complete NAV coverage for all schemes.

6. Expense ratios range from 0.55% to 1.64%.

7. All schemes support SIP investments starting from ₹500.

8. Minimum lumpsum investment ranges from ₹100 to ₹5000.

9. The launch_date column is currently stored as object datatype and may require conversion to datetime during preprocessing.

Conclusion:
The datasets show good data quality with complete scheme coverage, no missing values, and no AMFI code mismatches. The data is suitable for further ETL, SQL modelling, and exploratory data analysis.


In [31]:
aum = pd.read_csv(r"C:\Users\WINDOWS\OneDrive\Documents\Mutual_Fund_Analytics_Project\data\raw\03_aum_by_fund_house.csv")

In [32]:
aum.head()

,date,fund_house,aum_lakh_crore,aum_crore,num_schemes
0,2022-03-31,SBI Mutual Fund,6.05,605000,186
1,2022-03-31,ICICI Prudential MF,4.65,465000,216
2,2022-03-31,HDFC Mutual Fund,4.35,435000,195
3,2022-03-31,Nippon India MF,2.70,270000,177
4,2022-03-31,Kotak Mahindra MF,2.70,270000,168


In [33]:
print(aum.shape)


(90, 5)


In [34]:
print(aum.dtypes)


date               object
fund_house         object
aum_lakh_crore    float64
aum_crore           int64
num_schemes         int64
dtype: object


In [35]:
aum.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90 entries, 0 to 89
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   date            90 non-null     object 
 1   fund_house      90 non-null     object 
 2   aum_lakh_crore  90 non-null     float64
 3   aum_crore       90 non-null     int64  
 4   num_schemes     90 non-null     int64  
dtypes: float64(1), int64(2), object(2)
memory usage: 3.6+ KB


In [36]:
aum.describe()

,aum_lakh_crore,aum_crore,num_schemes
count,90.000000,9.000000e+01,90.000000
mean,4.352889,4.352889e+05,152.200000
std,2.734328,2.734328e+05,52.108832
min,1.050000,1.050000e+05,56.000000
25%,2.525000,2.525000e+05,95.000000
50%,3.450000,3.450000e+05,172.500000
75%,5.675000,5.675000e+05,195.000000
max,12.500000,1.250000e+06,216.000000


In [37]:
aum["fund_house"].nunique()

10

In [38]:
aum["fund_house"].unique()

array(['SBI Mutual Fund', 'ICICI Prudential MF', 'HDFC Mutual Fund',
       'Nippon India MF', 'Kotak Mahindra MF', 'Aditya Birla Sun Life MF',
       'Axis Mutual Fund', 'UTI Mutual Fund', 'Mirae Asset MF',
       'DSP Mutual Fund'], dtype=object)

In [39]:
aum.isnull().sum()

date              0
fund_house        0
aum_lakh_crore    0
aum_crore         0
num_schemes       0
dtype: int64

observation:

Dataset contains quarterly AUM information for 10 major mutual fund houses.

No missing values were found.

AUM ranges from ₹1.05 lakh crore to ₹12.50 lakh crore.

The average AUM across fund houses is approximately ₹4.35 lakh crore.

The number of schemes managed by AMCs ranges from 56 to 216.

In [40]:
sip = pd.read_csv(r"C:\Users\WINDOWS\OneDrive\Documents\Mutual_Fund_Analytics_Project\data\raw\04_monthly_sip_inflows.csv")

In [41]:
sip.head()

,month,sip_inflow_crore,active_sip_accounts_crore,new_sip_accounts_lakh,sip_aum_lakh_crore,yoy_growth_pct
0,2022-01,11517,4.91,9.10,4.80,NaN
1,2022-02,11438,4.93,8.20,4.85,NaN
2,2022-03,12328,5.09,10.50,5.01,NaN
3,2022-04,11863,5.48,9.52,5.12,NaN
4,2022-05,12286,5.55,8.10,5.15,NaN


In [42]:
print(sip.shape)

(48, 6)


In [43]:
print(sip.dtypes)

month                         object
sip_inflow_crore               int64
active_sip_accounts_crore    float64
new_sip_accounts_lakh        float64
sip_aum_lakh_crore           float64
yoy_growth_pct               float64
dtype: object


In [44]:
sip.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48 entries, 0 to 47
Data columns (total 6 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   month                      48 non-null     object 
 1   sip_inflow_crore           48 non-null     int64  
 2   active_sip_accounts_crore  48 non-null     float64
 3   new_sip_accounts_lakh      48 non-null     float64
 4   sip_aum_lakh_crore         48 non-null     float64
 5   yoy_growth_pct             36 non-null     float64
dtypes: float64(4), int64(1), object(1)
memory usage: 2.4+ KB


In [45]:
sip.describe()

,sip_inflow_crore,active_sip_accounts_crore,new_sip_accounts_lakh,sip_aum_lakh_crore,yoy_growth_pct
count,48.000000,48.000000,48.000000,48.000000,36.000000
mean,19577.520833,7.189583,9.893750,8.632083,31.456944
std,6354.329621,1.271457,5.370745,3.613579,11.768190
min,11438.000000,4.910000,7.500000,4.800000,15.800000
25%,13658.500000,6.110000,8.775000,5.790000,20.245000
50%,18224.000000,7.150000,9.200000,7.265000,28.295000
75%,25944.250000,8.300000,9.562500,10.525000,40.807500
max,31002.000000,9.350000,46.000000,15.900000,53.050000


In [46]:
sip.isnull().sum()

month                         0
sip_inflow_crore              0
active_sip_accounts_crore     0
new_sip_accounts_lakh         0
sip_aum_lakh_crore            0
yoy_growth_pct               12
dtype: int64

In [47]:
sip["sip_inflow_crore"].max()

np.int64(31002)

In [48]:
sip["sip_inflow_crore"].min()

np.int64(11438)

observation:

1. Dataset contains 48 monthly observations from 2022–2025.

2. No missing values were found except for 12 expected null values in yoy_growth_pct.

3. SIP inflows increased from ₹11,438 crore to ₹31,002 crore during the analysis period.

4. Active SIP accounts grew from 4.91 crore to 9.35 crore.

5. Monthly new SIP registrations reached as high as 46 lakh accounts.

6. Average year-over-year growth was approximately 31.46%.

7. The data indicates strong growth in retail participation in mutual fund investments.

In [49]:
category_inflows = pd.read_csv(r"C:\Users\WINDOWS\OneDrive\Documents\Mutual_Fund_Analytics_Project\data\raw\05_category_inflows.csv")

In [50]:
category_inflows.head()

,month,category,net_inflow_crore
0,2024-04,Large Cap,2413.0
1,2024-04,Mid Cap,3897.0
2,2024-04,Small Cap,3533.0
3,2024-04,Flexi Cap,4947.0
4,2024-04,Large & Mid Cap,4214.0


In [55]:
print(category_inflows.shape)

(144, 3)


In [54]:
print(category_inflows.dtypes)

month                object
category             object
net_inflow_crore    float64
dtype: object


In [56]:
category_inflows.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 144 entries, 0 to 143
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   month             144 non-null    object 
 1   category          144 non-null    object 
 2   net_inflow_crore  144 non-null    float64
dtypes: float64(1), object(2)
memory usage: 3.5+ KB


In [58]:
category_inflows.describe()

,net_inflow_crore
count,144.000000
mean,6473.881944
std,9718.335947
min,437.000000
25%,1801.250000
50%,4148.000000
75%,5304.000000
max,41952.000000


In [59]:
category_inflows.isnull().sum()

month               0
category            0
net_inflow_crore    0
dtype: int64

In [60]:
category_inflows["category"].value_counts()

category
Large Cap            12
Mid Cap              12
Small Cap            12
Flexi Cap            12
Large & Mid Cap      12
ELSS                 12
Value/Contra         12
Sectoral/Thematic    12
Liquid               12
Short Duration       12
Gilt                 12
Hybrid               12
Name: count, dtype: int64

In [61]:
category_inflows.groupby("category")["net_inflow_crore"].sum().sort_values(ascending=False)

category
Liquid               451275.0
Sectoral/Thematic    103829.0
Flexi Cap             63989.0
Large & Mid Cap       57752.0
Short Duration        55530.0
Mid Cap               55312.0
Small Cap             46596.0
Hybrid                38868.0
Large Cap             25633.0
Value/Contra          16980.0
Gilt                  10395.0
ELSS                   6080.0
Name: net_inflow_crore, dtype: float64

1. Dataset contains 144 observations across 12 mutual fund categories.

2. No missing values were found.

3. Liquid Funds attracted the highest net inflow of ₹4,51,275 crore.

4. Sectoral/Thematic Funds and Flexi Cap Funds were among the most preferred equity categories.

5. Small Cap Funds received significantly higher inflows than Large Cap Funds, indicating stronger investor appetite for growth-oriented investments.

6. ELSS funds recorded the lowest net inflows among the categories analyzed.

In [63]:
folio = pd.read_csv(r"C:\Users\WINDOWS\OneDrive\Documents\Mutual_Fund_Analytics_Project\data\raw\06_industry_folio_count.csv")


In [64]:
folio.head()

,month,total_folios_crore,equity_folios_crore,debt_folios_crore,hybrid_folios_crore,others_folios_crore
0,2022-01,13.26,9.28,1.86,0.80,1.33
1,2022-04,13.91,9.74,1.95,0.83,1.39
2,2022-07,13.85,9.69,1.94,0.83,1.38
3,2022-10,14.12,9.88,1.98,0.85,1.41
4,2023-01,14.81,10.37,2.07,0.89,1.48


In [65]:
print(folio.shape)

(21, 6)


In [66]:
print(folio.dtypes)

month                   object
total_folios_crore     float64
equity_folios_crore    float64
debt_folios_crore      float64
hybrid_folios_crore    float64
others_folios_crore    float64
dtype: object


In [68]:
folio.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21 entries, 0 to 20
Data columns (total 6 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   month                21 non-null     object 
 1   total_folios_crore   21 non-null     float64
 2   equity_folios_crore  21 non-null     float64
 3   debt_folios_crore    21 non-null     float64
 4   hybrid_folios_crore  21 non-null     float64
 5   others_folios_crore  21 non-null     float64
dtypes: float64(5), object(1)
memory usage: 1.1+ KB


In [69]:
folio.describe()

,total_folios_crore,equity_folios_crore,debt_folios_crore,hybrid_folios_crore,others_folios_crore
count,21.000000,21.000000,21.000000,21.000000,21.000000
mean,19.829524,13.880476,2.776190,1.190000,1.983333
std,4.583376,3.207799,0.640308,0.274864,0.459079
min,13.260000,9.280000,1.860000,0.800000,1.330000
25%,15.540000,10.880000,2.180000,0.930000,1.550000
50%,19.980000,13.990000,2.800000,1.200000,2.000000
75%,23.890000,16.720000,3.340000,1.430000,2.390000
max,26.120000,18.280000,3.660000,1.570000,2.610000


In [70]:
folio.isnull().sum()

month                  0
total_folios_crore     0
equity_folios_crore    0
debt_folios_crore      0
hybrid_folios_crore    0
others_folios_crore    0
dtype: int64

1. Dataset contains 21 quarterly observations of mutual fund folio counts.

2. No missing values were found.

3. Total folios increased from 13.26 crore to 26.12 crore during the period.

4. Equity folios represent the largest share of investor accounts.

5. Debt and Hybrid folios remain significantly lower than Equity folios.

6. The data indicates strong growth in retail participation in mutual fund investments.

In [72]:
performance = pd.read_csv(r"C:\Users\WINDOWS\OneDrive\Documents\Mutual_Fund_Analytics_Project\data\raw\07_scheme_performance.csv")

In [73]:
performance.head()

,amfi_code,scheme_name,fund_house,category,plan,return_1yr_pct,return_3yr_pct,return_5yr_pct,benchmark_3yr_pct,alpha,beta,sharpe_ratio,sortino_ratio,std_dev_ann_pct,max_drawdown_pct,aum_crore,expense_ratio_pct,morningstar_rating,risk_grade
0,119551,SBI Bluechip Fund - Regular Plan - Growth,SBI Mutual Fund,Large Cap,Regular,12.42,12.36,14.45,11.49,0.87,0.89,0.88,1.29,14.0,-21.70,14288,1.54,4,Moderate
1,119552,SBI Bluechip Fund - Direct Plan - Growth,SBI Mutual Fund,Large Cap,Direct,15.25,11.30,14.23,9.52,1.78,0.87,0.81,1.29,14.0,-24.43,1231,0.66,3,Moderate
2,119598,SBI Small Cap Fund - Regular Plan - Growth,SBI Mutual Fund,Small Cap,Regular,24.56,23.39,20.67,22.16,1.23,0.89,0.94,1.35,25.0,-13.35,19259,1.43,5,Very High
3,119599,SBI Small Cap Fund - Direct Plan - Growth,SBI Mutual Fund,Small Cap,Direct,20.59,23.14,21.82,22.01,1.13,1.04,0.93,1.67,25.0,-24.78,36061,0.72,4,Very High
4,119120,SBI Magnum Gilt Fund - Regular Plan - Growth,SBI Mutual Fund,Gilt,Regular,5.34,6.07,5.43,4.47,1.60,0.22,1.52,2.11,4.0,-2.30,24101,0.77,5,Low


In [74]:
print(performance.shape)

(40, 19)


In [75]:
print(performance.dtypes)

amfi_code               int64
scheme_name            object
fund_house             object
category               object
plan                   object
return_1yr_pct        float64
return_3yr_pct        float64
return_5yr_pct        float64
benchmark_3yr_pct     float64
alpha                 float64
beta                  float64
sharpe_ratio          float64
sortino_ratio         float64
std_dev_ann_pct       float64
max_drawdown_pct      float64
aum_crore               int64
expense_ratio_pct     float64
morningstar_rating      int64
risk_grade             object
dtype: object


In [77]:
performance.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 19 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   amfi_code           40 non-null     int64  
 1   scheme_name         40 non-null     object 
 2   fund_house          40 non-null     object 
 3   category            40 non-null     object 
 4   plan                40 non-null     object 
 5   return_1yr_pct      40 non-null     float64
 6   return_3yr_pct      40 non-null     float64
 7   return_5yr_pct      40 non-null     float64
 8   benchmark_3yr_pct   40 non-null     float64
 9   alpha               40 non-null     float64
 10  beta                40 non-null     float64
 11  sharpe_ratio        40 non-null     float64
 12  sortino_ratio       40 non-null     float64
 13  std_dev_ann_pct     40 non-null     float64
 14  max_drawdown_pct    40 non-null     float64
 15  aum_crore           40 non-null     int64  
 16  expense_ra

In [76]:
performance.describe()

,amfi_code,return_1yr_pct,return_3yr_pct,return_5yr_pct,benchmark_3yr_pct,alpha,beta,sharpe_ratio,sortino_ratio,std_dev_ann_pct,max_drawdown_pct,aum_crore,expense_ratio_pct,morningstar_rating
count,40.000000,40.000000,40.000000,40.000000,40.000000,40.000000,40.000000,40.000000,40.000000,40.000000,40.000000,40.00000,40.000000,40.000000
mean,120247.000000,14.376000,14.089000,14.516750,12.835500,1.253500,0.873250,1.361750,2.082500,14.962500,-19.200250,26091.60000,1.237000,4.250000
std,14534.998667,4.883023,4.617253,4.454021,4.740972,0.447412,0.224846,1.475805,2.203144,6.669282,8.819164,13809.11134,0.386584,0.742484
min,100016.000000,4.260000,5.140000,5.430000,3.960000,0.510000,0.220000,0.800000,1.030000,0.500000,-33.500000,979.00000,0.550000,3.000000
25%,118632.750000,11.735000,12.035000,12.340000,10.690000,0.887500,0.890000,0.865000,1.270000,14.000000,-25.062500,17400.50000,0.787500,4.000000
50%,119551.500000,14.620000,14.205000,14.185000,13.090000,1.205000,0.960000,0.925000,1.445000,14.000000,-20.600000,26713.00000,1.425000,4.000000
75%,120842.250000,16.392500,15.882500,17.585000,14.775000,1.700000,1.000000,0.985000,1.637500,19.000000,-14.255000,38125.00000,1.540000,5.000000
max,149324.000000,24.930000,23.390000,23.800000,22.160000,1.980000,1.040000,7.680000,10.370000,25.000000,-2.230000,49046.00000,1.640000,5.000000


In [78]:
performance.isnull().sum()

amfi_code             0
scheme_name           0
fund_house            0
category              0
plan                  0
return_1yr_pct        0
return_3yr_pct        0
return_5yr_pct        0
benchmark_3yr_pct     0
alpha                 0
beta                  0
sharpe_ratio          0
sortino_ratio         0
std_dev_ann_pct       0
max_drawdown_pct      0
aum_crore             0
expense_ratio_pct     0
morningstar_rating    0
risk_grade            0
dtype: int64

In [79]:
performance["risk_grade"].value_counts()

risk_grade
Moderate           16
High                8
Very High           6
Low                 6
Moderately High     4
Name: count, dtype: int64

In [80]:
performance["morningstar_rating"].value_counts()

morningstar_rating
5    17
4    16
3     7
Name: count, dtype: int64

In [81]:
#Top Performing Funds (5-Year Return)
performance.sort_values("return_5yr_pct",ascending=False)[["scheme_name","return_5yr_pct"]].head(10)

,scheme_name,return_5yr_pct
29,ABSL Small Cap Fund - Regular - Growth,23.80
27,Axis Small Cap Fund - Regular - Growth,22.62
17,Nippon India Small Cap Fund - Regular - Growth,21.88
3,SBI Small Cap Fund - Direct Plan - Growth,21.82
2,SBI Small Cap Fund - Regular Plan - Growth,20.67
39,DSP Small Cap Fund - Regular - Growth,20.61
38,DSP Midcap Fund - Regular - Growth,19.00
26,Axis Midcap Fund - Regular - Growth,18.94
21,Kotak Emerging Equity Fund - Regular - Growth,17.75
7,HDFC Mid-Cap Opportunities Fund - Regular - Gr...,17.69


In [82]:
#Highest Alpha
performance.sort_values("alpha",ascending=False)[["scheme_name","alpha"]].head(5)

,scheme_name,alpha
9,HDFC Short Term Debt Fund - Regular - Growth,1.98
21,Kotak Emerging Equity Fund - Regular - Growth,1.91
14,ICICI Pru Liquid Fund - Regular - Growth,1.85
22,Kotak Flexicap Fund - Regular - Growth,1.85
29,ABSL Small Cap Fund - Regular - Growth,1.84


In [83]:
#Highest sharpe ratio
performance.sort_values("sharpe_ratio", ascending=False)[["scheme_name","sharpe_ratio"]].head(5)

,scheme_name,sharpe_ratio
14,ICICI Pru Liquid Fund - Regular - Growth,7.68
23,Kotak Liquid Fund - Regular - Growth,6.18
30,ABSL Liquid Fund - Regular - Growth,5.14
9,HDFC Short Term Debt Fund - Regular - Growth,1.84
4,SBI Magnum Gilt Fund - Regular Plan - Growth,1.52


observation : 

1. Dataset contains performance metrics for 40 mutual fund schemes.

2. No missing values were found.

3. Average 5-year return across funds is approximately 14.52%.

4. Small Cap funds dominate the top-performing schemes based on 5-year returns.

5. HDFC Short Term Debt Fund recorded the highest Alpha (1.98).

6. ICICI Liquid Fund recorded the highest Sharpe Ratio (7.68), indicating strong risk-adjusted performance.

7. Most schemes are classified as Moderate risk and carry Morningstar ratings of 4 or 5 stars.

8. Different metrics identify different "best" funds, highlighting the importance of evaluating both returns and risk.

In [84]:
transactions = pd.read_csv(r"C:\Users\WINDOWS\OneDrive\Documents\Mutual_Fund_Analytics_Project\data\raw\08_investor_transactions.csv")

In [85]:
transactions.head()

,investor_id,transaction_date,amfi_code,transaction_type,amount_inr,state,city,city_tier,age_group,gender,annual_income_lakh,payment_mode,kyc_status
0,INV003054,2024-01-01,119092,SIP,1834,Telangana,Hyderabad,T30,56+,Female,77.1,UPI,Verified
1,INV002952,2024-01-01,148567,Redemption,392882,Punjab,Amritsar,B30,18-25,Male,7.1,Cheque,Verified
2,INV003420,2024-01-01,118636,SIP,912,Haryana,Faridabad,B30,36-45,Male,47.2,Mandate,Verified
3,INV003436,2024-01-01,118634,SIP,1102,Maharashtra,Mumbai,T30,36-45,Female,54.4,Cheque,Pending
4,INV004691,2024-01-01,119094,Lumpsum,8682,Delhi,Noida,T30,26-35,Male,14.5,Net Banking,Pending


In [86]:
print(transactions.shape)

(32778, 13)


In [87]:
print(transactions.dtypes)

investor_id            object
transaction_date       object
amfi_code               int64
transaction_type       object
amount_inr              int64
state                  object
city                   object
city_tier              object
age_group              object
gender                 object
annual_income_lakh    float64
payment_mode           object
kyc_status             object
dtype: object


In [88]:
transactions.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32778 entries, 0 to 32777
Data columns (total 13 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   investor_id         32778 non-null  object 
 1   transaction_date    32778 non-null  object 
 2   amfi_code           32778 non-null  int64  
 3   transaction_type    32778 non-null  object 
 4   amount_inr          32778 non-null  int64  
 5   state               32778 non-null  object 
 6   city                32778 non-null  object 
 7   city_tier           32778 non-null  object 
 8   age_group           32778 non-null  object 
 9   gender              32778 non-null  object 
 10  annual_income_lakh  32778 non-null  float64
 11  payment_mode        32778 non-null  object 
 12  kyc_status          32778 non-null  object 
dtypes: float64(1), int64(2), object(10)
memory usage: 3.3+ MB


In [89]:
transactions.describe()

,amfi_code,amount_inr,annual_income_lakh
count,32778.000000,32778.000000,32778.000000
mean,120264.617518,107437.318628,26.181219
std,14370.205345,150415.905084,20.805425
min,100016.000000,400.000000,3.000000
25%,118632.000000,3153.000000,10.600000
50%,119551.000000,17782.500000,19.700000
75%,120843.000000,189324.250000,37.400000
max,149324.000000,597498.000000,99.700000


In [91]:
transactions.isnull().sum()

investor_id           0
transaction_date      0
amfi_code             0
transaction_type      0
amount_inr            0
state                 0
city                  0
city_tier             0
age_group             0
gender                0
annual_income_lakh    0
payment_mode          0
kyc_status            0
dtype: int64

In [98]:
#Transaction Type Analysis
transactions["transaction_type"].value_counts()

transaction_type
SIP           19716
Lumpsum        8095
Redemption     4967
Name: count, dtype: int64

In [99]:
#Gender Analysis
transactions["gender"].value_counts()

gender
Male      21809
Female    10969
Name: count, dtype: int64

In [100]:
#City Tier Analysis
transactions["city_tier"].value_counts()

city_tier
T30    21719
B30    11059
Name: count, dtype: int64

In [101]:
#Payment Mode Analysis
transactions["payment_mode"].value_counts()

payment_mode
Net Banking    8250
Cheque         8228
UPI            8154
Mandate        8146
Name: count, dtype: int64

In [102]:
#Age Group Analysis
transactions["age_group"].value_counts()

age_group
26-35    13463
36-45     8146
18-25     4916
46-55     3779
56+       2474
Name: count, dtype: int64

In [106]:
#Investment Amount by Age Group
transactions.groupby("age_group")["amount_inr"].sum().sort_values(ascending=False)

age_group
26-35    1451600218
36-45     871647528
18-25     531639392
46-55     405406469
56+       261286823
Name: amount_inr, dtype: int64

In [105]:
#Investment Amount by Age Group
transactions.groupby("state")["amount_inr"].sum().sort_values(ascending=False).head(10)

state
Punjab            315780459
Tamil Nadu        315177237
Madhya Pradesh    308312493
Rajasthan         298645822
Gujarat           298358940
West Bengal       297182514
Telangana         290219284
Delhi             289633404
Uttar Pradesh     285368873
Haryana           279634354
Name: amount_inr, dtype: int64

observation:

1. Dataset contains 32,778 investor transactions.

2. No missing values were found.

3. SIP transactions account for approximately 60% of all transactions, making them the preferred investment mode.

4. Male investors contribute about two-thirds of total transactions.

5. Investors aged 26–35 form the largest customer segment.This group contributes more than ₹145 crore in transaction value and represents the largest    investor segment in the dataset.

6. T30 cities account for the majority of transactions, although B30 cities contribute significantly.

7. Payment mode usage is evenly distributed across UPI, Net Banking, Cheque, and Mandate channels.

8. The findings indicate that young urban investors are the primary drivers of mutual fund adoption.

In [107]:
holdings = pd.read_csv(r"C:\Users\WINDOWS\OneDrive\Documents\Mutual_Fund_Analytics_Project\data\raw\09_portfolio_holdings.csv")

In [108]:
holdings.head()

,amfi_code,stock_symbol,stock_name,sector,weight_pct,market_value_cr,current_price_inr,portfolio_date
0,119551,POWERGRID,Power Grid Corporation,Utilities,13.85,737.09,6011.08,2025-12-31
1,119551,HDFCBANK,HDFC Bank Ltd,Banking,11.19,88.97,1074.65,2025-12-31
2,119551,GRASIM,Grasim Industries Ltd,Diversified,9.90,208.45,5964.59,2025-12-31
3,119551,DRREDDY,Dr. Reddy's Laboratories,Pharma,4.76,161.32,3748.82,2025-12-31
4,119551,ASIANPAINT,Asian Paints Ltd,Paints,10.25,725.90,1321.45,2025-12-31


In [109]:
print(holdings.shape)

(322, 8)


In [110]:
print(holdings.dtypes)

amfi_code              int64
stock_symbol          object
stock_name            object
sector                object
weight_pct           float64
market_value_cr      float64
current_price_inr    float64
portfolio_date        object
dtype: object


In [111]:
holdings.isnull().sum()

amfi_code            0
stock_symbol         0
stock_name           0
sector               0
weight_pct           0
market_value_cr      0
current_price_inr    0
portfolio_date       0
dtype: int64

In [112]:
holdings["sector"].value_counts()

sector
Banking           60
IT                40
Pharma            38
Automobile        33
Utilities         24
Infrastructure    22
FMCG              21
Telecom           15
Diversified       14
Energy            13
Cement            12
NBFC              11
Paints            10
Consumer Goods     9
Name: count, dtype: int64

In [113]:
holdings["stock_name"].value_counts().head(10)

stock_name
Bharti Airtel Ltd          15
Grasim Industries Ltd      14
HCL Technologies Ltd       13
ICICI Bank Ltd             13
NTPC Ltd                   13
Reliance Industries Ltd    13
Adani Ports & SEZ Ltd      12
Axis Bank Ltd              12
UltraTech Cement Ltd       12
Tata Motors Ltd            12
Name: count, dtype: int64

In [114]:
holdings.groupby("sector")["market_value_cr"].sum().sort_values(ascending=False)

sector
Banking           62840.29
IT                38477.11
Pharma            34606.10
Automobile        34296.97
Utilities         25108.63
Infrastructure    22433.39
FMCG              21151.15
Telecom           16051.45
Energy            15286.54
Diversified       13897.79
Cement            11611.97
Paints            10612.07
Consumer Goods     9859.70
NBFC               8615.46
Name: market_value_cr, dtype: float64

In [116]:
holdings.sort_values( "weight_pct",ascending=False)[["stock_name","sector","weight_pct"]].head(10)

,stock_name,sector,weight_pct
183,Infosys Ltd,IT,38.18
229,Sun Pharmaceutical Industries,Pharma,35.07
118,Bharti Airtel Ltd,Telecom,30.44
245,NTPC Ltd,Utilities,28.95
276,State Bank of India,Banking,28.61
40,Grasim Industries Ltd,Diversified,28.25
64,Mahindra & Mahindra Ltd,Automobile,26.07
44,Wipro Ltd,IT,25.90
289,Bharti Airtel Ltd,Telecom,25.53
102,Sun Pharmaceutical Industries,Pharma,23.46


Portfolio Holdings Analysis:

1. Dataset contains 322 portfolio holding records with no missing values.

2. Banking is the most represented sector with 60 holdings.

3. Banking also attracts the highest market value allocation at ₹62,840 crore.

4. IT and Pharma are the second and third most preferred sectors.

5. Bharti Airtel, Grasim Industries, ICICI Bank, HCL Technologies and NTPC are among the most frequently held stocks.

6. Infosys, Sun Pharma and Bharti Airtel receive some of the highest portfolio allocations, indicating strong fund manager confidence.

7. The portfolio composition suggests that mutual funds are heavily positioned towards Banking, IT, Pharma and Automobile sectors.

In [117]:
benchmark = pd.read_csv(r"C:\Users\WINDOWS\OneDrive\Documents\Mutual_Fund_Analytics_Project\data\raw\10_benchmark_indices.csv")

In [118]:
benchmark.head()

,date,index_name,close_value
0,2022-01-03,NIFTY50,17492.79
1,2022-01-04,NIFTY50,17689.64
2,2022-01-05,NIFTY50,17835.05
3,2022-01-06,NIFTY50,17878.51
4,2022-01-07,NIFTY50,17759.15


In [119]:
print(benchmark.shape)

(8050, 3)


In [120]:
print(benchmark.dtypes)

date            object
index_name      object
close_value    float64
dtype: object


In [121]:
benchmark.isnull().sum()

date           0
index_name     0
close_value    0
dtype: int64

In [122]:
benchmark.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8050 entries, 0 to 8049
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   date         8050 non-null   object 
 1   index_name   8050 non-null   object 
 2   close_value  8050 non-null   float64
dtypes: float64(1), object(2)
memory usage: 188.8+ KB


In [123]:
benchmark.describe()


,close_value
count,8050.000000
mean,18351.950181
std,14169.995121
min,1444.130000
25%,2819.062500
50%,18392.960000
75%,26539.097500
max,79075.390000


observation:

The dataset contains historical benchmark index values used to compare mutual fund performance against market performance.

No missing values were found.

The benchmark dataset can later be used to:

Calculate market returns
Compare fund returns vs benchmark returns
Analyze market trends over time

Day 1 Data Quality Report
    
1. Loaded all 10 datasets successfully.

2. No critical missing values were found in:
   - Fund Master
   - NAV History
   - AUM by Fund House
   - Category Inflows
   - Industry Folio Count
   - Scheme Performance
   - Investor Transactions
   - Portfolio Holdings
   - Benchmark Indices

3. Monthly SIP Inflows contains 12 missing values in yoy_growth_pct.
   This is expected because YoY growth cannot be calculated for the initial months where prior-year comparison data is unavailable.

4. AMFI code validation was performed between fund_master and nav_history.

5. All AMFI codes present in fund_master were found in nav_history.

6. Dataset relationships are consistent and suitable for further analysis.

7. Date columns should be converted to datetime format during preprocessing.

8. No duplicate AMFI scheme codes were identified in the master dataset.